docs/rfcs/20220829-distributed-dynamic-routing.md

```diff
@@ -0,0 +1,224 @@
+                           │  └─────────────────────────┘   └─────────────────────────┘ │                 │         │
+                           │                                                            │                 │         │
+                           └────────────────────────────────────────────────────────────┘                 └─────────┘
+                                                                                                                     
+                                                                                                                     
+                                                                                                                     
+                                                                                                                     
+                                                                                                                     
+                                                                                                                     
+                                                                                                                     
+                                 ─                                                                                   
+```
+Explain：
+
+- Shard:  CeresDB Data Shard, One shard corresponds to a batch of tables. Shard's roles are divided into leader and follower, Only the leader is responsible for writing requests.
+- Node: CeresDB Compute Node, One node is responsible for processing multiple Shards.
+- Table:  Minimum unit of CeresDB scheduling, CeresDB implements distributed scheduling by adjusting the routing of the table.
+- Routing: Routing relationship from table to endpoint, The client relies on it to make requests to the correct node.
+- CeresMeta: An independently deployed service for managing and scheduling CeresDB cluster.
+
+### Metadata Definition
+#### ShardRole
+```rust
+pub enum ShardRole {
+    LEADER = 0, 
+    FOLLOWER = 1,
+    PENDING_LEADER = 2, 
+    PENDING_FOLLOWER = 3
+}
+
+// LEADER has RW permission
+// FOLLOWER has R permission
+// PENDING_LEADER is the follower who will become the leader in the process of leader switching，It has R permission
+// PENDING_FOLLOWER is the leader who will become the follower in the process of leader switching，It has RW permission
+```
+```
+┌────┐                   ┌────┐          
+│ RW ├─────────┐         │ RW ├─────────┐
+├────┘         │         ├────┘         │
+│    Leader    ◀─────────│PendingLeader │
+│              │         │              │
+└───────┬──────┘         └───────▲──────┘
+        │                        │       
+┌────┐  │                ┌────┐  │       
+│ R  ├──▼──────┐         │ R  ├──┴──────┐
+├────┘         │         ├────┘         │
+│   Pending    ├─────────▶   Follower   │
+│   Follower   │         │              │
+└──────────────┘         └──────────────┘
+```
```

Comment:
The comment and state machine are inconsistent on `PendingLeader` and `PendingFollower`'s permission.

Only `Leader` has RW permission, and all others are Read-only, right?